### PR TITLE
Tts component

### PR DESCRIPTION
**Pull request by vasystus**

### CODE REVIEW

**Review by mariaalouisaa — APPROVED** [review]

File: components/Styled-Components/TtsSpeakerAndText.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import styled from "styled-components";
```
Comment (PJSalter) [inline components/Styled-Components/TtsSpeakerAndText.js:1]: Great work on the styled component 🎨

File: components/tts.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    // props.children contains the text we want to be read aloud
+    const utterThis = new SpeechSynthesisUtterance(props.children);
+
+    // Slow the rate a bit.
+    utterThis.rate = 0.85;
+
+    // An empty array to hold the list of voices
+    let voicesUk = [];
+
+    function populateAndFilterVoiceList() {
```
Comment (PJSalter) [inline components/tts.js:22]: Awesome great work ✨

File: components/tts.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      // console.log(voicesUk);
+
+      // Hazel is the only item in the area. Set the voice to this
+      utterThis.voice = voicesUk[0];
+    }
+
+    populateAndFilterVoiceList();
+
+    // The list of voices is(/may be?) populated by the browser asynchronously.
```
Comment (PJSalter) [inline components/tts.js:39]: Very helpful use of comments, thank you so much ✨

File: components/tts.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    // The function to run when the TTS component is clicked:
+    function readOut() {
+      // Actually do the TTS! :)
+      synth.speak(utterThis);
+    }
+
+    const element = ref.current;
+
+    element.addEventListener("click", readOut);
```
Comment (PJSalter) [inline components/tts.js:57]: great to see a js event handler 😸

File: components/tts.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    function cleanup() {
+      element.removeEventListener("click", readOut);
+    }
+
+    return cleanup;
+  }, [props.children]);
+
+  return (
+    <TtsSpeakerAndText ref={ref}>
```
Comment (PJSalter) [inline components/tts.js:67]: good use of component name 🖌️ 🎨

File: package.json [hunk 6]
```diff
@@ -12,13 +12,14 @@
   "dependencies": {
     "next": "12.2.2",
     "react": "18.2.0",
-    "react-dom": "18.2.0"
+    "react-dom": "18.2.0",
+    "react-icons": "^4.4.0"
```
Comment (PJSalter) [inline package.json:16]: Thank you for installing the react icons ✨

**Review by PJSalter — APPROVED** [review]

**Pull request merged**